quipucords/api/scanjob/model.py

```diff
@@ -288,3 +299,62 @@ def validate_status_change(self, target_status, valid_current_status):
                          target_status, self.status)
             return True
         return False
+
+    def validate_extra_vars(self):
+        """Make sure that extra vars are a dictionary with boolean values.
+
+        :return extra_vars if no ValidationErrors are raised
+        """
+        extra_vars = self.get_optional_products_args()
+        if not isinstance(extra_vars, dict):
+            raise ValidationError(_(messages.SJ_EXTRA_VARS_DICT))
+        for key in extra_vars:
+            if not isinstance(extra_vars[key], bool):
+                raise ValidationError(_(messages.SJ_EXTRA_VARS_BOOL))
+            elif key not in [ScanJob.JBOSS_EAP,
+                             ScanJob.JBOSS_BRMS,
+                             ScanJob.JBOSS_FUSE]:
+                raise ValidationError(_(messages.SJ_EXTRA_VARS_KEY))
+        return extra_vars
+
```

Comment:
@chambridge here is where I do validation. I put it in the model because I felt that it could be used by the function below (line 329) and it seemed to make sense there.

Reply:
I think this code should be in the serializer as by this point in the code you have potentially already stored JSON that is not valid for our use and the Network inspect task may blow up with a `ValidationError` instead of the user seeing the issue on the POST.

Reply:
to be clear though it should be a function of the `ScanOptionsSerializer` class defined as `def validate_disable_optional_products(disable_optional_products):`